File: 4_counting_elements/frog_river_one.py
def solution(x, A):
    lst = [0] * x
    cnt = x
    l = len(A)
    for i in xrange(l):
        ind = A[i] - 1
        if lst[ind] == 0:
            lst[ind] = 1
            cnt -= 1
        if cnt == 0:
            return i
    return -1

File: 4_counting_elements/frog_river_one.cpp
#include <vector>

using namespace std;

int solution(int x, vector<int> &a) {
  int n = a.size();
  vector<int> leaves(x + 1, 0);
  int ret = 0;
  for (int i = 0; i < n; i++) {
    if (leaves[a[i]] == 0) {
      leaves[a[i]] = 1;
      ret++;
      if (ret == x)
        return i;
    }
  }
  return -1;
}
